src/creator.rs

```diff
@@ -69,7 +66,7 @@ impl<H: Hasher> Creator<H> {
             if round == 0 {
                 NodeMap::new_with_len(self.n_members)
             } else {
-                self.candidates_by_round[round - 1].clone()
+                self.candidates_by_round[(round - 1) as usize].clone()
```

Comment:
The problem is `u16` does not satisfy the `SliceIndex` trait, so it cannot be used directly anyway. `into()` doesn't work for the same reason – it cannot infer what type it should convert to, since something implementing `SliceIndex` is expected.

Defining `Round` as a new type might be overkill, since we only use this internally. Although I am definitely willing to do that if more people think it's a good idea.

Reply:
There is really no syntax for such ad-hoc type annotation in Rust? Declaring a new variable for it is kind of an overkill. At least Clippy checks `as` conversions. `usize::from` looks at least more type-safe(ish).

Reply:
New type is complete overkill in this scenario. We do upcast from `u16` to `usize` which are safe only to index a vector. I'd leave it as it is.